pkg/debugapi/topology.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package debugapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+)
+
+func (s *server) topologyHandler(w http.ResponseWriter, r *http.Request) {
+	ms, ok := s.TopologyDriver.(json.Marshaler)
+	if !ok {
+		s.Logger.Error("topology driver cast to json marshaler error")
+		jsonhttp.InternalServerError(w, "topology json marshal interface error")
+		return
+	}
```

Comment:
where do i do this? `init` function? also, this uses the assignment operator, doesn't this make `server.TopologyDriver` nil?

Reply:
Ups, sorry, I meant that the type of TopologyDriver in sever struct (Options to be more precise) can have the type of anon interface `interface{ topology.PeerAdder; json.Marshal }`. This will make the check for json.Marshaler on compile time and this runtime type check can be removed.

Reply:
nice trick. i tried to implement this but it is working too well, and i am getting compile time errors of topology driver not implementing the json marshaler interface, this is since i am trying to return interfaces directly on the constructors (for example when I return a new mock topology driver). not sure if to expose more concrete types or interfaces in this case. and in this case, i could just include the `json.Marshaler` interface directly on the topology driver interface and get it over with (and also eliminate the cast and the compile time interface check in the server)

Reply:
I see, yes, I think that it would be the most correct to add json.Marshaler to the interface directly. Thanks for looking into this.